### PR TITLE
Lib architecture and communication bases

### PR DESCRIPTION
**Pull request by matthiasrigaud**

### CODE REVIEW

File: lib/Config.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  socketLoop->run();
+  if (rv != SUCCESS)
+    throw std::exception();
+}
+
+///
+/// \todo implementation
+///
+LibConfig::Config::Config(std::string const &name)
+  : name(name)
```
Comment (Gaspard--) [inline lib/Config.cpp]: `Config::id` won't be initialised.

File: lib/Config.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+				  rv = SOCKET_NOT_FOUND;
+				else
+				  rv = CONNECTION_ERROR;
+				std::cout << "ERROR" << std::endl;
+			      });
+  socket->once<uvw::ConnectEvent>([&rv](const uvw::ConnectEvent&, uvw::PipeHandle&) {
+				    rv = SUCCESS;
+				    std::cout << "All succeed" << std::endl;
+				 });
+  std::cout << "Try to connect to " << socketPath << std::endl;
```
Comment (Gaspard--) [inline lib/Config.cpp]:
```suggestion
  std::cout << "Trying to connect to " << socketPath << std::endl;
```

File: lib/ReturnedValue.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+///
+
+#pragma once
+
+#ifdef __cplusplus
+namespace LibConfig
+{
+#endif
+
+  enum ReturnedValue
```
[the remark this answers is not in the document]
Comment (matthiasrigaud) [reply lib/ReturnedValue.h]: I thought it was better to have the returned values on another file, to have a best visibility. But maybe it's a bad idea

[the remark this answers is not in the document]
Comment (Gaspard--) [reply lib/ReturnedValue.h]: I think it would have the opposite effect. People are more likely not to realise there is a separate file for error codes.

**Pull request merged**